src/test/regress/expected/multi_subquery.out

```diff
@@ -152,6 +152,147 @@ ERROR:  cannot push down this subquery
 DETAIL:  Limit in subquery without limit in the outermost query is unsupported
 -- reset the flag for next query
 SET citus.subquery_pushdown to OFF;
+-- some queries without a subquery uses subquery planner
+SELECT l_orderkey
+FROM
+	lineitem_subquery l
+JOIN
+	orders_subquery o
+ON (l_orderkey = o_orderkey)
+WHERE
+	(o_orderkey < l_quantity)
+ORDER BY l_orderkey DESC
+LIMIT 10;
+ l_orderkey 
+------------
+         39
+         39
+         39
+         39
+         38
+         37
+         37
+         37
+         36
+         33
+(10 rows)
+
+-- query is not supported if does not have equi-join clause
+SELECT l_orderkey
+FROM
+	lineitem_subquery l
+JOIN
+	orders_subquery o
+ON (l_orderkey < o_orderkey)
+WHERE
+	(o_orderkey < l_quantity)
+ORDER BY l_orderkey DESC
+LIMIT 10;
+ERROR:  cannot pushdown the query since all relations are not joined using distribution keys
+DETAIL:  Each relation should be joined with at least one another relation using distribution keys and equality operator.
+-- distinct queries work
+SELECT DISTINCT l_orderkey
+FROM
+	lineitem_subquery l
+JOIN
+	orders_subquery o
+ON (l_orderkey = o_orderkey)
+WHERE
+	(o_orderkey < l_quantity)
+ORDER BY l_orderkey DESC
+LIMIT 10;
+ l_orderkey 
+------------
+         39
+         38
+         37
+         36
+         33
+         32
+          7
+          6
+          5
+          4
+(10 rows)
+
+-- count(distinct) queries work
+SELECT COUNT(DISTINCT l_orderkey)
+FROM
+	lineitem_subquery l
+JOIN
+	orders_subquery o
+ON (l_orderkey = o_orderkey)
+WHERE
+	(o_orderkey < l_quantity);
```

Comment:
it might be worth trying a few different cases that `IsJoinClause` prevents,

e.g. `o_orderkey - l_orderkey > 0`, `o_orderkey + 1 = l_orderkey`

Reply:
I am hitting
```
ERROR:  cannot perform local joins that involve expressions
DETAIL:  local joins can be performed between columns only
```

When there is a single join clause (i.e. `o_orderkey + 1 = l_orderkey`) whereas it throws another error message if we have (`o_orderkey + 1 = l_orderkey and o_orderkey < l_quantity`)
```
ERROR:  complex joins are only supported when all distributed tables are joined on their partition columns
```

Reply:
Should we push down those cases as well?

Reply:
done

Reply:
> Also maybe some kind of function(o.a,o.b)
 
Agreed. 

Also, maybe add explicit coercions as well such as (o_orderkey::bigint < l_quantity::int)

Reply:
For the joins check an example here: https://github.com/citusdata/citus/pull/1827/files#diff-0dcc934517678392319b744e5aa005c9R2043  
The first query is not supported, the next one is supported.

Reply:
adding that does not provide any additional value, but added.